main/inc/lib/webservices/Rest.php

```diff
@@ -1590,4 +1595,335 @@ private function encodeParams(array $additionalParams = [])
 
         return $encoded;
     }
+
+    /**
+     * @param $modelSessionId
+     * @param $sessionName
+     * @param $startDate
+     * @param $endDate
+     * @param array $extraFields
+     * @return array
+     * @throws Exception
+     */
+    public function createSessionFromModel($modelSessionId, $sessionName, $startDate, $endDate, array $extraFields = [])
+    {
+        if (!SessionManager::isValidId($modelSessionId)) {
+            throw new Exception(get_lang('ModelSessionDoesNotExist'));
+        }
+
+        $modelSession = SessionManager::fetch($modelSessionId);
+
+        if (api_is_multiple_url_enabled()) {
+            if (api_get_current_access_url_id() != -1) {
+                $modelSession['accessUrlId'] = api_get_current_access_url_id();
+            }
+        }
+
+        $newSessionId = SessionManager::create_session(
+            $sessionName,
+            $startDate,
+            $endDate,
+            $startDate,
+            $endDate,
+            $startDate,
+            $endDate,
+            $modelSession['coachId'],
+            $modelSession['sessionCategoryId'],
+            $modelSession['visibility'],
+            false,
+            $modelSession['duration'],
+            $modelSession['description'],
+            $modelSession['showDescription'],
+            $extraFields,
+            $modelSession['sessionAdminId'],
+            $modelSession['sendSubscriptionNotification'],
+            $modelSession['accessUrlId']
+        );
+
+        if (empty($newSessionId)) {
+            throw new Exception(get_lang('SessionNotRegistered'));
+        }
+
+        $promotionId = $modelSession['promotion_id'];
+        if ($promotionId) {
+            $sessionList = array_keys(SessionManager::get_all_sessions_by_promotion($promotionId));
+            $sessionList[] = $newSessionId;
+            SessionManager::subscribe_sessions_to_promotion($modelSession['promotion_id'], $sessionList);
+        }
+
+        $modelExtraFields = [];
+        $fields = SessionManager::getFilteredExtraFields($modelSessionId);
+        foreach($fields as $field) {
```

Comment:
We *always* use spaces after a loop or conditional operator and before the following parenthesis. Please fix here and on line 1660 (ideally, set that up in your code editor to avoid the issue later on).